### DISCUSSION TITLE
Self-hosted runner registered as idle but not picking up jobs

### DISCUSSION THREAD

**Discussion by consideRatio** [community/community#26823] [Actions]
<p>There is a runner available and Idle, and GitHub knows it has the label <code>rpi</code>.<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/6/b/6b420ccc7624e308c6a3f91d52fdd14513e0bf51.png" data-download-href="/uploads/short-url/fiQBftiTwl8eWRzbL3595H9yEbD.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/3837114/181100051-b04dd1ad-e0eb-46f8-a2fe-afeb200ab9d4.png" alt="image" data-base62-sha1="fiQBftiTwl8eWRzbL3595H9yEbD" width="690" height="301" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/6/b/6b420ccc7624e308c6a3f91d52fdd14513e0bf51_2_690x301.png, https://aws1.discourse-cdn.com/github/optimized/3X/6/b/6b420ccc7624e308c6a3f91d52fdd14513e0bf51_2_1035x451.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/6/b/6b420ccc7624e308c6a3f91d52fdd14513e0bf51.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/6/b/6b420ccc7624e308c6a3f91d52fdd14513e0bf51_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1129×493 44.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>There is a job started, and it says…</p>
<blockquote>
<p>Can’t find any online and idle self-hosted runner in the current repository, account/organization that matches the required labels: ‘rpi’</p>
<p>Waiting for a self-hosted runner to pickup this job…</p>
</blockquote>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/d/a/dadf5cfce308e389211a07ead5027ef1867e3c68.png" data-download-href="/uploads/short-url/veeDO9qZgLNVqAVwzoD1qgvkYK4.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/3837114/181100052-5999b504-1519-450b-994d-95d14c12abb3.png" alt="image" data-base62-sha1="veeDO9qZgLNVqAVwzoD1qgvkYK4" width="690" height="164" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/d/a/dadf5cfce308e389211a07ead5027ef1867e3c68_2_690x164.png, https://aws1.discourse-cdn.com/github/original/3X/d/a/dadf5cfce308e389211a07ead5027ef1867e3c68.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/d/a/dadf5cfce308e389211a07ead5027ef1867e3c68.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/d/a/dadf5cfce308e389211a07ead5027ef1867e3c68_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">988×235 18.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<h3>
<a name="question-1" class="anchor" href="#question-1"></a>Question</h3>
<p>Isn’t GitHub’s contradicting itself then - a bug with GitHub’s Runner management system? It is reporting an idle runner with the label <code>rpi</code>, but it is also reporting from a job in the organization that there are no runners matching that label.</p>
<h3>
<a name="about-my-runner-2" class="anchor" href="#about-my-runner-2"></a>About my runner</h3>
<p>My runner is deployed on a Raspberry Pi within a Kubernetes cluster, using the Helm chart <code>actions-runner-controller</code>. I’ve looked for issues with that setup for several days <a href="https://github.com/actions-runner-controller/actions-runner-controller/issues/732" rel="noopener nofollow ugc">but failed to conclude anything to be wrong</a>.</p>
<p>With no better signal of a failure besides GitHub reporting <code>Can't find any online and idle self-hosted runner in the current repository, account/organization that matches the required labels: 'rpi'</code>, which seems like a contradiction, I hope for help to pinpoint the issue. Is GitHub acting up, if not, what could be going wrong?</p>
<p>My runner is reporting the following:</p>
<pre><code class="lang-auto">Starting Runner listener with startup type: service
Started listener process
Started running service

√ Connected to GitHub

2021-08-29 05:34:16Z: Listening for Jobs
</code></pre>
<p>The actual runner software is from the Docker image <code>summerwind/actions-runner:v2.280.3-ubuntu-20.04-5b4b656</code> as provided by this project: <a href="https://github.com/actions-runner-controller/actions-runner-controller" class="inline-onebox" rel="noopener nofollow ugc">GitHub - actions-runner-controller/actions-runner-controller: Kubernetes controller for GitHub Act</a>. This mean that the <a href="https://github.com/actions/runner" class="inline-onebox" rel="noopener nofollow ugc">GitHub - actions/runner: The Runner for GitHub Actions</a> software is <code>v2.280.3</code> - the currently latest version, released 10 days ago.</p>
<h3>
<a name="about-my-github-org-repo-3" class="anchor" href="#about-my-github-org-repo-3"></a>About my GitHub org / repo</h3>
<p>My attempts are made within the <a href="https://github.com/SundellOpenSource" class="inline-onebox" rel="noopener nofollow ugc">Sundell Open Source Consulting AB · GitHub</a> org.</p>
<h4>
<a name="my-workflow-in-my-test-repohttpsgithubcomsundellopensourcetest-self-hosted-gha-runnersblobb033c88fa7116e9eb5eefc94750adfbfe50cc929githubworkflowstest-workflowyml-4" class="anchor" href="#my-workflow-in-my-test-repohttpsgithubcomsundellopensourcetest-self-hosted-gha-runnersblobb033c88fa7116e9eb5eefc94750adfbfe50cc929githubworkflowstest-workflowyml-4"></a>My workflow in <a href="https://github.com/SundellOpenSource/test-self-hosted-gha-runners/blob/b033c88fa7116e9eb5eefc94750adfbfe50cc929/.github/workflows/test-workflow.yml" rel="noopener nofollow ugc">my test repo</a>
</h4>
<pre><code class="lang-yaml">name: Test workflow

on:
  push:
  pull_request:
  workflow_dispatch:

jobs:
  build:
    runs-on: rpi

    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-python@v2
      - name: Run a one-line script
        run: echo Hello, world!
</code></pre>

**Comment by consideRatio** [comment]
<p>I used <code>tcpdump</code> to analyze the network traffic. I conclude that every 50 seconds, no matter if a job is started or not in a repository in my GitHub organization (<code>sundellopensource</code>), the same pattern will repeat every 50s as if my runner was never informed about any job it could take care of.</p>
<p>Note that the messages sent from 13.107.42.16 has the same length at all time, so its very likely its just a keep-alive kind of message without dynamic amounts of information about one or more jobs available.</p>
<p>From this, I conclude that there is no network issues, and the issue is that GitHub doesn’t try ask if my runner is ready to do work like indicated by the logs saying that no runners was found.</p>
<pre><code class="lang-auto">05:47:39.361567 IP 13.107.42.16.https &gt; sundellopensource-org-ks9dk-thqqx.54066: Flags [P.], seq 4445:5080, ack 14946, win 2053, length 635
05:47:39.366429 IP sundellopensource-org-ks9dk-thqqx.54066 &gt; 13.107.42.16.https: Flags [P.], seq 14946:17081, ack 5080, win 501, length 2135
05:47:39.379042 IP 13.107.42.16.https &gt; sundellopensource-org-ks9dk-thqqx.54066: Flags [.], ack 17081, win 2053, length 0

05:48:29.561837 IP 13.107.42.16.https &gt; sundellopensource-org-ks9dk-thqqx.54066: Flags [P.], seq 5080:5715, ack 17081, win 2053, length 635
05:48:29.566228 IP sundellopensource-org-ks9dk-thqqx.54066 &gt; 13.107.42.16.https: Flags [P.], seq 17081:19216, ack 5715, win 501, length 2135
05:48:29.580135 IP 13.107.42.16.https &gt; sundellopensource-org-ks9dk-thqqx.54066: Flags [.], ack 19216, win 2053, length 0

05:49:19.747976 IP 13.107.42.16.https &gt; sundellopensource-org-ks9dk-thqqx.54066: Flags [P.], seq 5715:6350, ack 19216, win 2053, length 635
05:49:19.752065 IP sundellopensource-org-ks9dk-thqqx.54066 &gt; 13.107.42.16.https: Flags [P.], seq 19216:21351, ack 6350, win 501, length 2135
05:49:19.764754 IP 13.107.42.16.https &gt; sundellopensource-org-ks9dk-thqqx.54066: Flags [.], ack 20656, win 2053, length 0
05:49:19.769739 IP 13.107.42.16.https &gt; sundellopensource-org-ks9dk-thqqx.54066: Flags [.], ack 21351, win 2050, length

05:50:09.951034 IP 13.107.42.16.https &gt; sundellopensource-org-ks9dk-thqqx.54066: Flags [P.], seq 6350:6985, ack 21351, win 2050, length 635
05:50:09.955472 IP sundellopensource-org-ks9dk-thqqx.54066 &gt; 13.107.42.16.https: Flags [P.], seq 21351:23486, ack 6985, win 501, length 2135
05:50:09.967411 IP 13.107.42.16.https &gt; sundellopensource-org-ks9dk-thqqx.54066: Flags [.], ack 23486, win 2053, length 0

05:51:00.138275 IP 13.107.42.16.https &gt; sundellopensource-org-ks9dk-thqqx.54066: Flags [P.], seq 6985:7620, ack 23486, win 2053, length 635
05:51:00.148926 IP sundellopensource-org-ks9dk-thqqx.54066 &gt; 13.107.42.16.https: Flags [P.], seq 23486:25621, ack 7620, win 501, length 2135
05:51:00.160910 IP 13.107.42.16.https &gt; sundellopensource-org-ks9dk-thqqx.54066: Flags [.], ack 25621, win 2053, length 0

05:51:50.342807 IP 13.107.42.16.https &gt; sundellopensource-org-ks9dk-thqqx.54066: Flags [P.], seq 7620:8255, ack 25621, win 2053, length 635
05:51:50.347999 IP sundellopensource-org-ks9dk-thqqx.54066 &gt; 13.107.42.16.https: Flags [P.], seq 25621:27756, ack 8255, win 501, length 2135
05:51:50.360267 IP 13.107.42.16.https &gt; sundellopensource-org-ks9dk-thqqx.54066: Flags [.], ack 27061, win 2053, length 0
05:51:50.360307 IP 13.107.42.16.https &gt; sundellopensource-org-ks9dk-thqqx.54066: Flags [.], ack 27756, win 2050, length 0

05:52:40.764346 IP 13.107.42.16.https &gt; sundellopensource-org-ks9dk-thqqx.54066: Flags [P.], seq 8255:8890, ack 27756, win 2050, length 635
05:52:40.770056 IP sundellopensource-org-ks9dk-thqqx.54066 &gt; 13.107.42.16.https: Flags [P.], seq 27756:29891, ack 8890, win 501, length 2135
05:52:40.781346 IP 13.107.42.16.https &gt; sundellopensource-org-ks9dk-thqqx.54066: Flags [.], ack 29196, win 2053, length 0
05:52:40.781435 IP 13.107.42.16.https &gt; sundellopensource-org-ks9dk-thqqx.54066: Flags [.], ack 29891, win 2050, length 0

05:53:30.952444 IP 13.107.42.16.https &gt; sundellopensource-org-ks9dk-thqqx.54066: Flags [P.], seq 8890:9525, ack 29891, win 2050, length 635
05:53:30.957273 IP sundellopensource-org-ks9dk-thqqx.54066 &gt; 13.107.42.16.https: Flags [P.], seq 29891:32026, ack 9525, win 501, length 2135
05:53:30.969253 IP 13.107.42.16.https &gt; sundellopensource-org-ks9dk-thqqx.54066: Flags [.], ack 31331, win 2053, length 0
05:53:30.975126 IP 13.107.42.16.https &gt; sundellopensource-org-ks9dk-thqqx.54066: Flags [.], ack 32026, win 2050, length 0

05:54:21.154476 IP 13.107.42.16.https &gt; sundellopensource-org-ks9dk-thqqx.54066: Flags [P.], seq 9525:10160, ack 32026, win 2050, length 635
05:54:21.158916 IP sundellopensource-org-ks9dk-thqqx.54066 &gt; 13.107.42.16.https: Flags [P.], seq 32026:34161, ack 10160, win 501, length 2135
05:54:21.171325 IP 13.107.42.16.https &gt; sundellopensource-org-ks9dk-thqqx.54066: Flags [.], ack 34161, win 2053, length 0
</code></pre>

**Comment by gautamkrishnar** [comment]
<aside class="quote no-group" data-username="consideRatio" data-post="1" data-topic="198240">
<div class="title">
<div class="quote-controls"></div>
 consideRatio:</div>
<blockquote>
<pre><code class="lang-auto">jobs:
  build:
    runs-on: rpi
</code></pre>
</blockquote>
</aside>
<p>Can you try using the following:</p>
<pre><code class="lang-auto">jobs:
  build:
    runs-on: [self-hosted,rpi]
</code></pre>
<p>Please let us know if it’s not still working</p>

**Comment by consideRatio** [comment]
<blockquote>
<p>Can’t find any online and idle self-hosted runner in the current repository, account/organization that matches the required labels: ‘self-hosted , rpi’</p>
<p>Waiting for a self-hosted runner to pickup this job…</p>
</blockquote>
<p>No difference in the results =/</p>

**Comment by ChristopherHX** [comment, marked as answer]
<p><strong>It’s a security feature of github action</strong><br>
The reason why your runner get’s no job is that public repositores are excluded by your runner group see your own screenshot of your first post.<br>
<code>All repositories, excluding public repositories</code> the default of github.</p>
<p>Exactly same problem <a href="https://github.com/ChristopherHX/github-act-runner/discussions/40#discussioncomment-1067366" class="inline-onebox" rel="noopener nofollow ugc">No runner matching the specified labels was found: self-hosted · Discussion #40 · ChristopherHX/github-act-runner · GitHub</a> with an instruction to allow public repositories.</p>

**Comment by consideRatio** [comment]
<p>Aaaaaaaaaaaaaaaaaaaaaahhhhhhhhhhhhhhhhhhhhhhhhhhhhhh… That was it, thank you.</p>
<p>Now I’m having that feeling that you have after loads of hours spent debugging something to conclude you have need to check a checkbox with one mouse click.</p>

**Comment by vinaysandela** [comment]
<p>Having same error on Internal repos in our Org.  Works fine for Private repos.<br>
There are no public allowed in our Enterprise level.<br>
Where should I enable for using for Internal repos ?</p>

**Reply by jeff-tilton** [reply]
@vinaysandela did you resolve this issue?

**Reply by divyanshu-reverie** [reply]
@jeff-tilton did you find anything ?

**Reply by abdushkur** [reply]
any updates? our repos also not public, but runners still not picking up jobs
BTW, I could not wait till someone give us a solution, so i removed the self hosted runner and added again, now it is picking up the jobs
